packages/types/src/utility.ts

```diff
@@ -55,6 +55,10 @@ export type UtilityConfig = {
   [property in LiteralUnion<CssProperty>]?: PropertyConfig
 }
 
-export type ExtendableUtilityConfig = UtilityConfig & {
-  extend?: UtilityConfig
+type UtilityConfigWithExtend = {
+  [pattern in LiteralUnion<CssProperty>]?: PropertyConfig | UtilityConfig | undefined
+}
+
+export type ExtendableUtilityConfig = UtilityConfigWithExtend & {
+  extend?: UtilityConfig | undefined
 }
```

Comment:
I'm confused about this type

Reply:
before:
<img width="1416" alt="Screenshot 2023-09-27 at 13 52 36" src="https://github.com/chakra-ui/panda/assets/47224540/94a81b98-5582-41d6-b405-e0317b20f791">

after:
<img width="1588" alt="Screenshot 2023-09-27 at 13 53 16" src="https://github.com/chakra-ui/panda/assets/47224540/7a219197-b8d5-4d62-807e-b4f997322f38">

I'm guessing this is needed for TS to understand what values are allowed

this is actually closer to what's done in the other extendable types like `ExtendablePatterns`:
https://github.com/chakra-ui/panda/blob/fix/1430-extendable-utility/packages/types/src/config.ts#L69-L72

Reply:
the `extend` key will then constraint the mapped type